drivers/platform/stm32/stm32_gpio.c

```diff
@@ -112,6 +112,15 @@ static int32_t _gpio_init(struct no_os_gpio_desc *desc,
 	else
 		return -EINVAL;
 
+	if (!param->extra) {
+		pextra = (struct stm32_gpio_init_param*)malloc(sizeof(*pextra));
```

Comment:
this is not ok. desc->extra needs to exist, it is used in all other API functions

better drop this commit and leave the stm32_init_param stuff as mandatory.

Reply:
desc->extra exists, the only thing that could not exist is stm32_init_param.